### PR TITLE
Adding Deployment API - For Review

### PR DESCRIPTION
**Pull request by ericduran**
This is still a work in progress but I figure I'll open up this PR now for comments, reviews, concern, etc..

### CODE REVIEW

**Comment by lol768** [comment]
@ericduran Are you open to contributions to this PR? Would be happy to look into submitting a PR or two to https://github.com/ericduran/php-github-api/tree/deploymentapi in order to add some unit tests.

**Comment by ericduran** [comment]
@lol768 sure, I'm getting back into this next week so if you can get to it before me awesome :) currently distracted with the status api. Ha.

**Comment by lol768** [comment]
@ericduran Awesome! I've opened a PR to your fork over (here)[https://github.com/ericduran/php-github-api/pull/1], let me know your thoughts there. The status API does look pretty neat.

**Comment by ericduran** [comment]
@lol768 ha just updated to the latest master it already has the status api, looking at the PR now.

**Comment by ericduran** [comment]
@lol768 Ha, thanks for all that clean up and well everything. So we can keep the same PR and the conversation going I just went ahead and added you as a collab on my repo so if you need to make more changes based on any feedback you can just do that without requiring me to just merge PRs.

That being said this looks pretty good :+1: 

@Nek- thoughts? Would you like the commits cleans up? or is that fine? I like keeping the history since there are multiple contributors here.

**Comment by stof** [comment]
@ericduran if you want to squash commits to clean the history, only squash commits which belong to the same author. Don't squash together commits with different authorship

**Comment by lol768** [comment]
Merge conflicts resolved.

File: test/Github/Tests/Api/DeploymentTest.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+class DeploymentTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function shouldCreateDeployment()
+    {
+        $api = $this->getApiMock();
+        $deploymentData = array("ref" => "fd6a5f9e5a430dddae8d6a8ea378f913d3a766f9");
```
Comment (pilot) [inline test/Github/Tests/Api/DeploymentTest.php]: use single quotes

**Comment by lol768** [comment]
Updated per commit comment.

**Comment by lol768** [comment]
Thanks!

**Pull request merged**